variables.tf

```diff
@@ -0,0 +1,69 @@
+variable "prefix" {
+  description = "The prefix used for all resources in this environment"
+  type        = string
+
+  validation {
```

Comment:
Nice addition 😄 One alternative I've seen (or maybe used, but I can't remember why!) is to explicitly restrict it to a preset list, using the `contains` rule; something like:
```
  validation {
    condition     = contains(["valid_prefix_1", "prefix_2", "prefix_3"], var.prefix)
    error_message = "Valid values for var: prefix are (...)."
  } 
```